src/sentry/search/events/datasets/discover.py

```diff
@@ -1958,3 +1959,20 @@ def _event_type_filter_converter(self, search_filter: SearchFilter) -> WhereType
                 return None
 
         return self.builder.default_filter_converter(search_filter)
+
+    def _transaction_filter_converter(self, search_filter: SearchFilter) -> WhereType | None:
+        if self.builder.dataset == Dataset.Transactions:
+            operator = search_filter.operator
+            value = search_filter.value.value
+
+            if operator in ("=", "!=") and value == "":
+                # !has:transaction
+                if operator == "=":
+                    raise InvalidSearchQuery(
+                        "All events have a transaction so this query wouldn't return anything"
+                    )
+                else:
+                    # All events have a "transaction" since we map null -> unparam so no need to filter
+                    return None
```

Comment:
Copied this over from `metrics.py` and `metrics_layer.py` files. We have this restriction now on the transactions dataset because all events in that dataset will have the transaction field.

Reply:
It was used in a test so I figured we could just copy it over here too to maintain the behaviour.